### DISCUSSION TITLE
do not understand in the global state

### DISCUSSION THREAD

**Discussion by bobwatcherx** [lumeland/lume#264] [General]
I want to change data from data.xml
suppose
name = "jessie"
and when accessed .md and .tmpl.ts and other files the state has changed to jessie

where should I put this syntax in which file?

site.data("name", "jessie");

on the web: https://lume.land/docs/configuration/data/

thanks

**Comment by bobwatcherx** [comment]
in _config.ts , sorry noob 

thanks

**Comment by oscarotero** [comment]
In Lume the data is stored in `_data.*` files or `_data/*` folders. And their scope is the directory where the file saved is and subdirectories. There's a better explanation here: https://lume.land/docs/creating-pages/shared-data/

You can create a `_data.yml` file (or use other formats, like json, ts, js) with the data that you want to share with your pages:

```yml
# _data.yml

name: jessie
```

Placing this file in the root folder, will affects to all pages of your site. If you save it in a subdirectory, only affects to the pages in this subdirectory (and sub-subdirectories).

`site.data("name", "jessie")` is equivalent to create a `_data` file in the root folder, but it's more practical for plugins, so they can inject their data in the site, or if you don't want to create a file only to add a variable.

**Reply by bobwatcherx** [reply]
> In Lume the data is stored in `_data.*` files or `_data/*` folders. And their scope is the directory where the file saved is and subdirectories. There's a better explanation here: https://lume.land/docs/creating-pages/shared-data/
> 
> You can create a `_data.yml` file (or use other formats, like json, ts, js) with the data that you want to share with your pages:
> 
> ```yaml
> # _data.yml
> 
> name: jessie
> ```
> 
> Placing this file in the root folder, will affects to all pages of your site. If you save it in a subdirectory, only affects to the pages in this subdirectory (and sub-subdirectories).
> 
> `site.data("name", "jessie")` is equivalent to create a `_data` file in the root folder, but it's more practical for plugins, so they can inject their data in the site, or if you don't want to create a file only to add a variable.

the case is that I want the data changes to be on the client
example
```
_data.yml

name: jack
```

in home.jsx
```
 export default(data) => (
  <>
{data.name}
  <input type="text" onChange={(e)=> data.name(e.target.value)}/>
    
);
```

no state change
is there a way out

**Reply by oscarotero** [reply]
Ok, understood.
Lume is not a frontend framework but a sofware to create static sites at build time. It generates static files like html, css or javascript files.
You can use it to write server side scripts and use the `esbuild` plugin to compile them (there's an example in this repo: https://github.com/lumeland/react-todo). But Lume doesn't do any "magic" here.
